### PR TITLE
created Therapist model , improved the booking routes, updated the searching algo.

### PR DESCRIPTION
**Pull request by VikasChaudhary121**
// Created the Therapists model.
// Shifted the booking route into the users..
// moved the custom hook in the utils directory.

### CODE REVIEW

File: backend/models/Therapist/therapistModel.js [hunk 1]
```diff
@@ -1,33 +1,80 @@
+const mongoose = require("mongoose");
```
Comment (chandan24042001s) [inline backend/models/Therapist/therapistModel.js:2]: Why is this therapist model ? It is even not related to the Booking and even the therapist section itself.

File: backend/index.js [hunk 2]
```diff
@@ -45,26 +45,23 @@ app.get("/", (req, res) => {
 });
 app.use("/api/users", userRoute);
 app.use("/api/admin", adminRoute);
-app.use("/api/booking", bookingRoute);
-app.use("/api/therapist",therapistRoute);
+app.use("/api/therapist", therapistRoute);
```
Comment (chandan24042001s) [inline backend/index.js:48]: What the F ?  Do u want to work on therapist section ? This code is not making sense give reason .

Why u needed to create a route for therapist itself ?

**Comment by VikasChaudhary121** [comment]
Ayush told me to create the therapist model

File: backend/models/Therapist/therapistModel.js (hunk 1 above)
Comment (chandan24042001s) [reply backend/models/Therapist/therapistModel.js:2]: @VikasChaudhary121 this is too much details for user about the therapist.  Just fetch I'd and use name and corresponding expertise 


Make sense !

**Review by chandan24042001s — APPROVED** [review]

**Pull request merged**